### PR TITLE
Fix invalid date issue without catch or raise exception

### PR DESCRIPTION
**Pull request by halan**
This solve https://github.com/apotonick/reform/pull/267 without catch generic exceptions or any other. I use the same test at the other PR.

### CODE REVIEW

**Comment by PedroFelipe** [comment]
:+1:

**Comment by matheusazzi** [comment]
:+1:

**Comment by akaKuruma** [comment]
:+1:

**Comment by luizrogeriocn** [comment]
:+1:

**Comment by cefigueiredo** [comment]
:+1:

**Comment by brunoskonrad** [comment]
:+1:

**Comment by iagomoreira** [comment]
smooth :+1:

**Comment by bernardogfilho** [comment]
nice :+1:

**Comment by luizvarela** [comment]
:+1:

**Comment by romulostorel** [comment]
:+1:

**Comment by halan** [comment]
:+1:

File: lib/reform/form/multi_parameter_attributes.rb [hunk 1]
```diff
@@ -24,10 +24,12 @@ def call(params)
 
   private
     def params_to_date(year, month, day, hour, minute)
-      return nil if [year, month, day].any?(&:blank?)
+      if [year, month, day].any?(&:blank?) || !Date.valid_date?(*[year, month, day].map(&:to_i))
```
Comment (carloslopes) [inline lib/reform/form/multi_parameter_attributes.rb]: You are generating the same array twice, can you refactor and reuse the same?

**Comment by waldyr** [comment]
:-1:

File: lib/reform/form/multi_parameter_attributes.rb (hunk 1 above)
Comment (halan) [inline lib/reform/form/multi_parameter_attributes.rb]: I think about it, I just try solve the problem with minimum code, but I will do it, above have the same array too.

File: lib/reform/form/multi_parameter_attributes.rb [hunk 2]
```diff
@@ -24,10 +24,12 @@ def call(params)
...
+        return nil
+      end
 
       if hour.blank? && minute.blank?
```
Comment (carloslopes) [inline lib/reform/form/multi_parameter_attributes.rb]: if you are handling dates, why are you concerned about hour and minute?

File: lib/reform/form/multi_parameter_attributes.rb (hunk 1 above)
Comment (andrewhr) [inline lib/reform/form/multi_parameter_attributes.rb]: Blank strings become `0` when coerced via `to_i`. Given that any `0` is understood as an invalid date, what do you think of skipping the `blank?` check? Maybe I'm missing something?

**Comment by apotonick** [comment]
Haha, great but since when do we have this :+1: on my gems?! :laughing:

**Comment by apotonick** [comment]
Cool, so this catches invalid dates without an exception? That's great! Any concerns?

**Comment by halan** [comment]
No, it's ruby 1.8 compatible. It will return `nil` if receives invalid date, instead `ArgumentError` raised by `Date.new` or `Time.new`

**Comment by halan** [comment]
I just put the tests from #267.

**Comment by apotonick** [comment]
@halan But here you test invalid dates: https://github.com/apotonick/reform/pull/289/files#diff-ad5370b7e7280e5aeb58a9412a4ade3bR20 and I can't see any `rescue` in your code.

Please join https://gitter.im/trailblazer/chat for a quick chat! :beers:

**Comment by brunoskonrad** [comment]
:clap:

**Pull request merged**